docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
 
-### Changing package.json
-Use a unique `npm` name for your plugin. Change `package.json` name field as follows:
+## Updating `package.json`
+
+Change the name property in the `package.json` file to the plug-in name.
+
+Open the `package.json` file in a text editor and replace the name field with the following information:
 
 ```typescript
   "name": "@zowe/files-util",
 ```
 
-Issue the command `npm install` against the local repository.
+This tutorial uses `@zowe/files-util` as the tutorial plug-in name.
+
+**To create a unique plug-in:** Replace `@zowe/files-util` with a unique plug-in name. This allows you to publish the plug-in under that name to the `npm` registry in the future.
+
+## Adjusting Imperative CLI Framework configuration
+Define json configurations for the plug-in to Imperative.
+
+Change the `src/imperative.ts` file to contain the following configurations:
 
-### Adjusting Imperative CLI Framework configuration
-Change `imperative.ts` to contain the following:
 ```typescript
 import { IImperativeConfig } from "@zowe/imperative";
 
 const config: IImperativeConfig = {
     commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
     rootCommandDescription: "Files utility plugin for Zowe CLI",
-    envVariablePrefix: "FILES_UTIL_PLUGIN",
-    defaultHome: "~/.files_util_plugin",
     productDisplayName: "Files Util Plugin",
     name: "files-util"
 };
 
 export = config;
 ```
-Here we adjusted the description and other fields in the `imperative` JSON configuration to be relevant to this plug-in.
 
-### Adding third-party packages
+When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
 
-We'll use the following packages to create a programmatic API:
+**To create a unique plug-in:** **[what do we advise here?]**
 
-- `npm install --save diff`
-- `npm install -D @types/diff`
+## Adding third-party packages
+Install third-party packages as dependencies for the plug-in's client-side API.
 
+ Follow these steps:
 
-### Creating a Node.js programmatic API
-In `files-util/src/api`, create a file named `DataSetDiff.ts`. The content of `DataSetDiff.ts` should be the following:
-```typescript
-import { AbstractSession } from "@zowe/imperative";
-import { Download, IDownloadOptions, IZosFilesResponse } from "@zowe/cli";
-import * as diff from "diff";
-import { readFileSync } from "fs";
+1. Run `npm install --save-exact diff` to install the diff package (which includes methods for comparing text).
 
-export class DataSetDiff {
+2. Run `npm install --save-dev @types/diff` to install the typescript type definitions for the diff package as a [development dependency](https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file).
 
-    public static async diff(session: AbstractSession, oldDataSet: string, newDataSet: string) {
+    When successful, the `diff` and `@types/diff` packages are added to the dependency list in the `package.json` file.
```

Comment:
General question:
- Should the `when successful` sections have the same indentation level as the `to create a unique plugin` ?

Reply:
Not necessarily